qiskit_experiments/base_analysis.py

```diff
@@ -59,8 +92,14 @@ def run(self, experiment_data, save=True, return_figures=False, **options):
         # Wait for experiment job to finish
         # experiment_data.block_for_result()
 
+        # Get runtime analysis options
+        analysis_options = copy.copy(self.options)
+        analysis_options.update_options(**options)
```

Comment:
This was matching the current way options are handled for simulators. When a class is initialized it initializes a copy of the default options specified in the class design, and then everything in options is mutable. For analysis options it's currently coded so you can pass option values at construction `Analysis(**options)` which is equivalent to setting them after default construction, `Analysis().set_options(**options)`. These are equivalent. The `Analysis.run` method can also be called with options, which will override any stored values in `analysis.options` just for that execution, without changing any of the stored options.

However thinking about this more, since analysis options are also stored in experiment maybe this is overkill and we could remove the `set_options` ahd just have the default options and `run(**options)` to override them to keep the analysis clas more like a function without side effects (like it was before). I'll try make some changes to simplify this.

Reply:
Sounds good, thank you :)